app/src/main/java/com/sdpteam/connectout/chat/ChatDirectory.java

```diff
@@ -0,0 +1,30 @@
+package com.sdpteam.connectout.chat;
+
+import android.view.View;
+import android.widget.ListAdapter;
+
+import androidx.lifecycle.LiveData;
+
+import com.firebase.ui.database.FirebaseListOptions;
+
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+
+public interface ChatDirectory {
+
+    /**
+     * Save your new ChatMessage
+     */
+    void saveMessage(ChatMessage message);
+
+    String getProfileUserName();
+
+    void setUpListAdapter(Function<FirebaseListOptions.Builder, FirebaseListOptions.Builder> setLayout,
```

Comment:
There are some warnings about the raw use parameterized class 'FirebaseListOptions.Builder'. Can you check that?

Reply:
yup. I didn't put the <ChatMessage>